src/main/java/org/testobject/appium/junit/TestParser.java

```diff
@@ -8,39 +8,14 @@
 
 class TestParser {
 
-	private static final Pattern METHOD_AND_CLASS_NAME_PATTERN_STRICT = Pattern.compile("(.+)[\\[\\(](.*)[\\]\\)]");
-	private static final Pattern METHOD_AND_CLASS_NAME_PATTERN_LOOSE = Pattern.compile("(.+)");
-
-
 	public static Test from(Description testDescription) {
+		String className = testDescription.getClassName();
 
-		String className;
-		String methodName;
-		String deviceId = null;
-
-		Matcher matcher = METHOD_AND_CLASS_NAME_PATTERN_STRICT.matcher(testDescription.getMethodName());
```

Comment:
The method name, device id, and data center id should never have spaces, so this should work fine.